### PR TITLE
refactor: fix likes and refactor everything

### PR DESCRIPTION
**Pull request by Laetta**

### CODE REVIEW

File: src/stats/favorite-players/command-cooldown.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Player } from 'org.bukkit.entity';
+import * as yup from 'yup';
+import { dataType } from '../../common/datas/holder';
+import { dataView, deleteView } from '../../common/datas/view';
+
+const LikeCooldownData = dataType('playerLikeCooldownData', {
+  previousTime: yup.number().notRequired(),
+});
+
+const LOKE_COOLDOWN_HOURS = 12;
```
Comment (raikasdev) [inline src/stats/favorite-players/command-cooldown.ts:10]: Typo, should be LIKE_COOLDOWN_HOURS?

File: src/stats/favorite-players/command-cooldown.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  return remaining;
+}
+
+export function startLikeCooldown(player: Player) {
+  const view = dataView(LikeCooldownData, player);
+  view.previousTime = new Date().getTime();
+}
+
+const LIKE_COOLDOWN_MS = 1000 * 60 * 60 * LOKE_COOLDOWN_HOURS;
```
Comment (raikasdev) [inline src/stats/favorite-players/command-cooldown.ts:36]: Remember to change this too.

**Review by raikasdev — CHANGES_REQUESTED** [review]

File: src/stats/favorite-players/likes.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  return true;
+}
+
+/************************
+ * Private functions
+ ************************/
+
+const likesDb: Table<string, number> = getTable('likes-table');
```
Comment (bensku) [inline src/stats/favorite-players/likes.ts:52]: Databases (backed by H2 MVStore) support UUIDs natively as keys and values. They *probably* take less space than UUID strings. If this doesn't work, update craftjs-plugin types.

Also, no need to postfix with -table

File: src/stats/favorite-players/likes.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  });
+}
+
+function isValidPlayer(player: OfflinePlayer | null) {
+  if (!player) return false;
+  if (player.isOnline()) return true;
+  return player.hasPlayedBefore();
+}
+
+const likesRefreshedDb: Table<string, number> = getTable(
```
Comment (bensku) [inline src/stats/favorite-players/likes.ts:116]: Note to self: add API for persistent storage of single values. Table is good enough for now.

File: src/stats/favorite-players/messages.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
+import { Player } from 'org.bukkit.entity';
+import { color, text } from 'craftjs-plugin/chat';
+import { OfflinePlayer } from 'org.bukkit';
+import { t, addTranslation } from '../../common/localization/localization';
+import { PlayerLikes } from './PlayerLikes';
+import { millisecondsToDuration } from '../../common/helpers/duration';
+
+const yellow = (msg: string) => color('#FFFF55', text(msg));
```
Comment (bensku) [inline src/stats/favorite-players/messages.ts:8]: Should we introduce these as global functions?

File: src/stats/likes.ts [hunk 6]
```diff
@@ -0,0 +1,258 @@
+import { color, text } from 'craftjs-plugin/chat';
```
Comment (bensku) [inline src/stats/likes.ts:1]: Is this file a duplicate?

**Review by bensku — CHANGES_REQUESTED** [review]

**Review by bensku — APPROVED** [review]

**Pull request merged**